orttraining/orttraining/python/training/onnxblock/_graph_utils.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# _graph_utils.py
+
+import copy
+import onnx
+import random
+
+from onnxruntime.capi._pybind_state import GradientGraphBuilder
+
+
+def get_output_from_output_name(onnx_model, output_name):
+    """Returns the graph output given the output name"""
+
+    # Iterate over the graph outputs looking for output_name
+    for output in onnx_model.graph.output:
+        if output.name == output_name:
+            return output
+
+    raise LookupError(f"The provided output name {output_name} is not a graph output.")
+
+
+def get_random_number():
+    """Return a random number in the range 0, 1000."""
+
+    return random.randint(0, 1000)
+
+
+def generate_random_graph_name(token):
+    """Return a string that can be used in the graph as a graph attribute name."""
+
+    return f"{get_random_number()}.{token}"
+
+
+def build_gradient_graph(
+    accessor, user_args_requiring_grad, user_args_not_requiring_grad, output_names
```

Comment:
can these other params also be a member of accessor object since they are tied into the model anyway

Reply:
yeah, I feel the model accessor can evolve over time and incorporate more execution specific things, like parameter names and such. I think for now though, let us keep the API as is, unless it blocks us from performing some operation.
But definitely over time, we can update the API.